### PR TITLE
Add `parseProgress` helper function

### PR DESCRIPTION
**Pull request by mattt**
```js
import Replicate, { parseProgress } from "replicate";

const replicate = new Replicate({})

const output = await replicate.run(model, { input }, (prediction) => {
  console.log(parseProgress(prediction));
});
```

### CODE REVIEW

**Review by mattrothenberg — APPROVED** [review]

**Comment by arbajkhan140** [comment]
Photo editing

**Comment by arbajkhan140** [comment]
Photo editing

File: index.d.ts [hunk 1]
```diff
@@ -280,4 +280,10 @@ declare module "replicate" {
         },
     secret: string
   ): boolean;
+
+  export function parseProgress(logs: Prediction | string): {
```
Comment (mattt) [inline index.d.ts:284]: How's the name? 

I had `parsePredictionProgress`, but that felt too wordy. Also, too specific to predictions (but maybe trainings should have a different method?)

Do you think it's worth mentioning logs in the method name?

**Review by zeke — COMMENTED** [review]
Curious about what models this will work for. Is there a semi-standardized way of producing progress output from models?

I see this is documented in code, but this should have public docs in the README too.

File: lib/util.js [hunk 2]
```diff
@@ -246,4 +246,38 @@ function isPlainObject(value) {
...
+  const lines = logs.split("\n").reverse();
+
+  for (const line of lines) {
+    const matches = line.match(pattern);
+
+    if (matches && matches.length === 4) {
+      return {
+        percentage: parseInt(matches[1], 10) / 100,
+        current: parseInt(matches[2], 10),
+        total: parseInt(matches[3], 10),
```
[the remark this answers is not in the document]
Comment (mattt) [reply lib/util.js:270]: Yes, the latter.

**Comment by mattt** [comment]
> Is there a semi-standardized way of producing progress output from models?

My understanding is that formatting comes from [tqdm](https://github.com/tqdm/tqdm). It's not a listed dependency in Cog, but used by other popular packages, including from [🤗 Diffusers](https://huggingface.co/docs/diffusers/main/en/api/logging):

> By default, tqdm progress bars are displayed during model download. logging.disable_progress_bar and logging.enable_progress_bar are used to enable or disable this behavior.

File: index.d.ts (hunk 1 above)
Comment (mattt) [reply index.d.ts:284]: Yeah, I think that's clearer. Going with that.

File: lib/util.js [hunk 3]
```diff
@@ -246,4 +246,38 @@ function isPlainObject(value) {
   );
 }
 
-module.exports = { transformFileInputs, validateWebhook, withAutomaticRetries };
+/**
+ * Parse progress from prediction logs.
+ * @param {object|string} input - A prediction object or string.
+ * @returns {object} - An object with the percentage, current, and total.
```
[the remark this answers is not in the document]
Comment (mattt) [reply lib/util.js:252]: Yessir!
